homeassistant/components/hyperion/sensor.py

```diff
@@ -0,0 +1,215 @@
+        hyperion_client: client.HyperionClient,
+        entity_description: SensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        self.entity_description = entity_description
+        self._device_id = get_hyperion_device_id(server_id, instance_num)
+        self._instance_name = instance_name
+        self._client = hyperion_client
+        self._state = "unknown"
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._device_id)},
+            manufacturer=HYPERION_MANUFACTURER_NAME,
+            model=HYPERION_MODEL_NAME,
+            name=self._instance_name,
+            configuration_url=self._client.remote_url,
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return server availability."""
+        return bool(self._client.has_loaded_state)
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks when entity added to hass."""
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                SIGNAL_ENTITY_REMOVE.format(self._attr_unique_id),
+                functools.partial(self.async_remove, force_remove=True),
+            )
+        )
+
+        self._client.add_callbacks(self._client_callbacks)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Cleanup prior to hass removal."""
+        self._client.remove_callbacks(self._client_callbacks)
+
+
+class HyperionVisibleEffectSensor(HyperionSensor):
+    """Class that displays the visible effect of a Hyperion instance."""
+
+    _attr_extra_state_attributes = {}
+
+    def __init__(
+        self,
+        server_id: str,
+        instance_num: int,
+        instance_name: str,
+        hyperion_client: client.HyperionClient,
+        entity_description: SensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+
+        super().__init__(
+            server_id, instance_num, instance_name, hyperion_client, entity_description
+        )
+
+        self._attr_unique_id = _sensor_unique_id(
+            server_id, instance_num, TYPE_HYPERION_SENSOR_VISIBLE_PRIORITY
+        )
+
+        self._client_callbacks = {
+            f"{KEY_PRIORITIES}-{KEY_UPDATE}": self._update_priorities
+        }
+
+    @callback
+    def _update_priorities(self, _: dict[str, Any] | None = None) -> None:
+        """Update Hyperion priorities."""
+        state_value = "unknown"
+        attrs = {}
+
+        for priority in self._client.priorities or []:
+            if KEY_VISIBLE in priority:
+                if priority[KEY_VISIBLE] is True:
+                    # Determine state
+                    state_value = ""
+                    if priority[KEY_COMPONENTID] == "COLOR":
+                        state_value = priority[KEY_VALUE][KEY_RGB]
+                    else:
+                        state_value = priority[KEY_OWNER]
+
+                    # Update extra attributes
```

Comment:
Nit: Comment is not really conveying anything that is not obvious from the code.